cmd/csi-attacher/main.go

```diff
@@ -109,7 +109,12 @@ func main() {
 			glog.Error(err.Error())
 			os.Exit(1)
 		}
-		if !supportsAttach {
+		supportsService, err := csiConn.SupportsPluginControllerService(ctx)
+		if err != nil {
+			glog.Error(err.Error())
+			os.Exit(1)
+		}
+		if !supportsAttach || !supportsService {
```

Comment:
I think you should check for `SupportsPluginControllerService` first and only when the driver supports controller service you can invoke controller methods such as `ControllerGetCapabilities` used in `SupportsControllerPublish`.